requests_oauthlib/oauth2_session.py

```diff
@@ -64,19 +64,15 @@ def __init__(self, client_id=None, client=None, auto_refresh_url=None,
         :param kwargs: Arguments to pass to the Session constructor.
         """
         super(OAuth2Session, self).__init__(**kwargs)
-        self.client_id = client_id
-        if client is not None and not self.client_id:
-            self.client_id = client.client_id
+        self._client = client or WebApplicationClient(client_id, token=token)
+        self.token = token or {}
```

Comment:
I'm confused -- I don't see the backwards-incompatibility, myself. Can you provide a concrete example demonstrating the behavior that changes in this pull request? An actual automated test would be awesome, but even just a few lines of Python in a comment would be helpful.